applications/Chat/examples/train_rlhf_sft.py

```diff
@@ -0,0 +1,230 @@
+import argparse
+import math
+import warnings
+
+import torch
+import torch.distributed as dist
+from coati.dataset import PromptDataset, SupervisedDataset
+from coati.models.bloom import BLOOMActor
+from coati.models.chatglm import ChatGLMActor
+from coati.models.chatglm.chatglm_tokenizer import ChatGLMTokenizer
+from coati.models.gpt import GPTActor
+from coati.models.llama import LlamaActor
+from coati.models.opt import OPTActor
+from coati.trainer import SFTTrainer
+from coati.trainer.strategies import DDPStrategy, GeminiStrategy, LowLevelZeroStrategy
+from datasets import load_dataset
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+from transformers import AutoTokenizer, BloomTokenizerFast, LlamaTokenizer
+from transformers.models.gpt2.tokenization_gpt2 import GPT2Tokenizer
+from transformers.trainer import get_scheduler
+
+from colossalai.logging import get_dist_logger
+from colossalai.nn.optimizer import HybridAdam
+
+# please use the same format as the training data of your reward model
+PROMPT_DICT = {
+    "prompt_input": (
+        "{instruction} {input}\n"
+    ),
+    "prompt_no_input": (
+        "{instruction}\n"
+    ),
+}
+
+def train(args):
+    # configure strategy
+    if args.strategy == "ddp":
+        strategy = DDPStrategy()
+    elif args.strategy == "colossalai_gemini":
+        strategy = GeminiStrategy(placement_policy="auto")
+    elif args.strategy == "colossalai_zero2":
+        strategy = LowLevelZeroStrategy(stage=2, placement_policy="cuda")
+    elif args.strategy == "colossalai_zero2_cpu":
+        strategy = LowLevelZeroStrategy(stage=2, placement_policy="cpu")
+    else:
+        raise ValueError(f'Unsupported strategy "{args.strategy}"')
+
+    # configure model
+    if args.lora_rank > 0:
+        warnings.warn("Gradient checkpoint is disabled when using LoRA")
+        args.grad_checkpoint = False
+    with strategy.model_init_context():
+        if args.model == "bloom":
+            model = BLOOMActor(pretrained=args.pretrain, lora_rank=args.lora_rank, checkpoint=args.grad_checkpoint)
```

Comment:
The concept of `actor` is only meaningful in RLHF stage. We can use the original LM here, and use booster api directly.